crates/api_models/src/events/payment.rs

```diff
@@ -108,6 +123,23 @@ impl ApiEventMetric for PaymentsRequest {
     }
 }
 
+#[cfg(feature = "v2")]
+impl ApiEventMetric for PaymentsCreateIntentRequest {
+    fn get_api_event_type(&self) -> Option<ApiEventsType> {
+        Some(ApiEventsType::Miscellaneous)
```

Comment:
Make this EventType as `Payment` instead of `Miscellaneous`

Reply:
we will not have the id to be populated in order to propogate this event as Payment. Instead should this be None? What if the merchant passes his reference id. Should we have that instead? Just a thought can be noted down to be fixed later

Reply:
Response-wise we can add payment id audit, then its fine to pass None.